api/v1alpha1/portal_webhook.go

```diff
@@ -28,36 +28,48 @@ import (
 // log is for logging in this package.
 var portallog = logf.Log.WithName("portal-resource")
 
-func (r *Portal) SetupWebhookWithManager(mgr ctrl.Manager) error {
+func (p *Portal) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
-		For(r).
+		For(p).
+		WithDefaulter(p).
+		WithValidator(p).
 		Complete()
 }
 
+//+kubebuilder:webhook:path=/mutate-core-kubebb-k8s-com-cn-v1alpha1-portal,mutating=true,failurePolicy=fail,sideEffects=None,groups=core.kubebb.k8s.com.cn,resources=portals,verbs=create;update,versions=v1alpha1,name=mportal.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomDefaulter = &Portal{}
+
+func (p *Portal) Default(ctx context.Context, obj runtime.Object) error {
+	portallog.Info("default", "name", p.Name)
+
+	return nil
+}
+
 //+kubebuilder:webhook:path=/validate-core-kubebb-k8s-com-cn-v1alpha1-portal,mutating=false,failurePolicy=fail,sideEffects=None,groups=core.kubebb.k8s.com.cn,resources=portals,verbs=create;update,versions=v1alpha1,name=vportal.kb.io,admissionReviewVersions=v1
```

Comment:
```suggestion
//+kubebuilder:webhook:path=/validate-core-kubebb-k8s-com-cn-v1alpha1-portal,mutating=false,failurePolicy=fail,sideEffects=None,groups=core.kubebb.k8s.com.cn,resources=portals,verbs=create;update;delete,versions=v1alpha1,name=vportal.kb.io,admissionReviewVersions=v1
```

If we want use Portal.ValidateDelete() and need run `make generate && make manifests`

Reply:
Good point! `Delete` should also be vallidated.